R/cohensFleissKappa.R

```diff
@@ -192,3 +193,59 @@ cohensFleissKappa <- function(jaspResults, dataset, options) {
   }
   return(jaspTable)
 }
+
+.computeKrippendorffsAlphaTable <- function(dataset, options, ready) {
+  # Create the JASP Table
+  jaspTable <- createJaspTable(title = gettextf("Krippendorff's alpha"))
+  jaspTable$addColumnInfo(name = "method", title = gettext("Method"), type = "string")
+  jaspTable$addColumnInfo(name = "kAlpha", title = gettext("Krippendorff's alpha"), type = "number")
+  jaspTable$position <- 2
+  
+  #dependencies
+  jaspTable$dependOn(
+    options = c(
+      "variables",
+      "krippendorffsAlpha",
+      "kappaIntervalOn",
+      "kappaConfidenceIntervalValue"
+    )
+  )
+  
+  formattedCIPercent <- format(
+    100 * options[["kappaConfidenceIntervalValue"]],
+    digits = 3,
+    drop0trailing = TRUE
+  )
+  
+  if (ready) {
+    #calculate Krippendorff's alpha
+    kAlphaData <- t(as.matrix(dataset))
+    method <- options[["alphaMethod"]]
+    kAlpha <- irr::kripp.alpha(kAlphaData, method)
+    
+    tableData <- list("method" = paste0(toupper(substr(method, 1, 1)), substr(method, 2, nchar(method))),
+                      "kAlpha" = kAlpha$value)
+    footnote <- gettextf('%i subjects/items and %i raters/measurements.', kAlpha$subjects, kAlpha$raters)
+    
+    if (options[["kappaIntervalOn"]]) {
+      bootstrapAlphas <- boot::boot(dataset, alpha.boot, R = 1000, method = method)
+      CIs <- boot::boot.ci(bootstrapAlphas, type = "perc", conf = options[["kappaConfidenceIntervalValue"]])
```

Comment:
```suggestion
n <- nrow(dataset)
alphas <- numeric(1e3)
for (i in seq_len(1e3)) {
  bootData <- as.matrix(dataset[sample.int(n, size = n, replace = TRUE), ])
  alphas[i] <- irr::kripp.alpha(t(boot_data), method = method)$value
}
conf <- options[["kappaConfidenceIntervalValue"]]
confs <- (1 + c(-conf, conf)) / 2
CIs <- quantile(alphas, probs = confs)

```
this way we could avoid having to use the boot package. The speed is almost equivalent.